internal/engine/upper.go

```diff
@@ -133,6 +133,9 @@ func (u Upper) Init(ctx context.Context, action InitAction) error {
 }
 
 func upperReducerFn(ctx context.Context, state *store.EngineState, action store.Action) {
+	if state.FatalError != nil {
```

Comment:
do you need to be able to handle the hud.ExitAction even in the error state? The DumpEngineStateAction might be good too

Reply:
ah good point!